File: leetcode/go/628.maximum-product-of-three-numbers.go
func maximumProduct(nums []int) int {
    n := len(nums)
    slices.Sort(nums)
    x, y := nums[0]*nums[1]*nums[n-1], nums[n-1] * nums[n-2] * nums[n-3]
    if x > y {
        return x
    }
    return y
}

File: leetcode/rust-leetcode/src/628.maximum-product-of-three-numbers.rs
impl Solution {
    pub fn maximum_product(mut nums: Vec<i32>) -> i32 {
        nums.sort();
        let mut max = 0;
        let l = nums.len();
        let mut max = nums[0] * nums[1] * nums[l-1];
        max = max.max(nums[l-1] * nums[l-2] * nums[l-3]);
        max
    }
}

